### PR TITLE
4211- DotNet Development for Messaging on SQS

### PR DESCRIPTION
**Pull request by alexanderwjrussell**
4211-DotNet Development for Messaging on SQS

#### 📲 What

Create a simple NuGet package which is a wrapper around AWS SDK for SQS.

#### 🤔 Why

In order for teams to utilise the stacks-cqrs-events templates on AWS they will need access to AWS SQS and this will resolve the configuration.

#### 🛠 How

Teams will be able to implement the NuGet package within their repositories and just attach a QueueUrl in their appsettings.json and the package will sort out the rest of the configuration.

#### 👀 Evidence

Screenshot of messages in SQS from a quick POC:
image

Example event:
image

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,89 @@
...
+
+#### 2.3.1 Event
+In this case the `NotifyEvent` has a `NotifyEventHandler`. The handler implements
+`Amido.Stacks.Application.CQRS.ApplicationEvents.IApplicationEventHandler<NotifyCommand, bool>` and the command implements
+`Amido.Stacks.Application.CQRS.ApplicationEvents.IApplicationEvent` interfaces.
+
+***NotifyEvent.cs***
+
+```cs
+   public class NotifyEvent : IApplicationEvent
```
Comment (PavelDanov) [inline README.md]: Can we use an event we have in the app like `MenuCreated`? The `NotifyEvent` is from the ASB package.

File: src/Amido.Stacks.SQS/Consumer/EventConsumer.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        private readonly IAmazonSQS _queueClient;
+        private readonly IOptions<AwsSqsConfiguration> _configuration;
+        private readonly ISecretResolver<string> _secretResolver;
+
+        public EventConsumer(
+            IOptions<AwsSqsConfiguration> configuration,
+            ISecretResolver<string> secretResolver,
+            IAmazonSQS queueClient)
+        {
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
```
Comment (PavelDanov) [inline src/Amido.Stacks.SQS/Consumer/EventConsumer.cs]: We generally use `this` instead of the underscore approach.

File: README.md (hunk 1 above)
Comment (alexanderwjrussell) [reply README.md]: Yeah, that makes sense. Thanks :)

File: src/Amido.Stacks.SQS/Consumer/EventConsumer.cs (hunk 2 above)
Comment (alexanderwjrussell) [reply src/Amido.Stacks.SQS/Consumer/EventConsumer.cs]: Thanks for letting me know. Will try to remember going forwards :)

**Review by PavelDanov — APPROVED** [review]

**Pull request merged**